### ISSUE TITLE
OpenGL is locked at 60 making audio choppy for WS games

### ISSUE THREAD

**Issue by Yasand123** [higan-emu/ares#720]
**Describe the bug**
OpenGL backend is locked at 60 for WS, v3.2 and v2.0. Other backends are _not_ affected. This happens despite both Synchronization and GPU sync being unchecked in the Video section.

Audio sync doesn't affect this behavior.

**To Reproduce**
1. Load --> Bandai --> WonderSwan ...
2. Make sure you're on the OpenGL backend
3. Game runs at 60

**Expected behavior**
Framerate should be unlocked (or Synced to audio if that's checked).

**Screenshots**

**Additional context**
- Arch Linux.
- Ares v129.
- Nvidia 2060.
- Monitor's refreshrate is actually 75, so systemwide Vsync would lock FPS at 75 which is what happens in other graphical applications.
- Installed via AUR package `ares-emu`

**Comment by LukeUsher** [comment]
I am unable to reproduce this on either Windows (11), Linux (Ubuntu), or macOS (Catalina).

If GPU sync is disabled, ares sets the glSwapInterval to 0, however, it only uses glXSwapIntervalMESA glXSwapIntervalSGI, there is no support for glXSwapIntervalEXT.

Perhaps your driver/gpu requires the use of glXSwapIntervalEXT?

**Comment by LukeUsher** [comment]
Can you check if the following patch fixes the issue for you? 

```patch
Index: ruby/video/glx2.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruby/video/glx2.cpp b/ruby/video/glx2.cpp
--- a/ruby/video/glx2.cpp	(revision 972495df89353a73bee91a3fe917853f8733d515)
+++ b/ruby/video/glx2.cpp	(date 1662454187865)
@@ -62,7 +62,7 @@
 
   auto setBlocking(bool blocking) -> bool override {
     acquireContext();
-    if(glXSwapInterval) glXSwapInterval(blocking);
+    glXSwapInterval(blocking);
     releaseContext();
     return true;
   }
@@ -300,10 +300,11 @@
     _glXContext = glXCreateContext(_display, visual, 0, GL_TRUE);
     glXMakeCurrent(_display, _glXWindow = _window, _glXContext);
 
-    if(!glXSwapInterval) glXSwapInterval = (int (*)(int))glGetProcAddress("glXSwapIntervalMESA");
-    if(!glXSwapInterval) glXSwapInterval = (int (*)(int))glGetProcAddress("glXSwapIntervalSGI");
+    glXSwapIntervalEXT  = (int (*)(Display*, GLXDrawable drawable, int))glGetProcAddress("glXSwapIntervalEXT");
+    glXSwapIntervalMESA = (int (*)(int))glGetProcAddress("glXSwapIntervalMESA");
+    glXSwapIntervalSGI  = (int (*)(int))glGetProcAddress("glXSwapIntervalSGI");
 
-    if(glXSwapInterval) glXSwapInterval(self.blocking);
+    glXSwapInterval(self.blocking);
 
     s32 value = 0;
     glXGetConfig(_display, visual, GLX_DOUBLEBUFFER, &value);
@@ -402,5 +403,15 @@
   u32 _glHeight = 0;
   u32 _glFormat = GL_UNSIGNED_INT_8_8_8_8_REV;
 
-  auto (*glXSwapInterval)(int) -> int = nullptr;
+  auto glXSwapInterval(int blocking) -> int {
+    //note that the ordering is very important! MESA declares SGI, but the SGI function does nothing
+    if(glXSwapIntervalEXT  && glXSwapIntervalEXT(_display, glXGetCurrentDrawable(), blocking)) return 1;
+    if(glXSwapIntervalMESA && glXSwapIntervalMESA(blocking)) return 1;
+    if(glXSwapIntervalSGI  && glXSwapIntervalSGI(blocking)) return 1;
+    return 0;
+  }
+
+  auto (*glXSwapIntervalMESA)(int) -> int = nullptr;
+  auto (*glXSwapIntervalSGI)(int) -> int = nullptr;
+  auto (*glXSwapIntervalEXT)(Display *dpy, GLXDrawable drawable, int interval) -> int = nullptr;
 };
Index: ruby/video/glx.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/ruby/video/glx.cpp b/ruby/video/glx.cpp
--- a/ruby/video/glx.cpp	(revision 972495df89353a73bee91a3fe917853f8733d515)
+++ b/ruby/video/glx.cpp	(date 1662454331255)
@@ -48,7 +48,7 @@
 
   auto setBlocking(bool blocking) -> bool override {
     acquireContext();
-    if(glXSwapInterval) glXSwapInterval(blocking);
+    glXSwapInterval(blocking);
     releaseContext();
     return true;
   }
@@ -246,9 +246,9 @@
     glXMakeCurrent(_display, _glXWindow = _window, _glXContext);
 
     //glXSwapInterval is used to toggle Vsync
-    //note that the ordering is very important! MESA declares SGI, but the SGI function does nothing
-    if(!glXSwapInterval) glXSwapInterval = (int (*)(int))glGetProcAddress("glXSwapIntervalMESA");
-    if(!glXSwapInterval) glXSwapInterval = (int (*)(int))glGetProcAddress("glXSwapIntervalSGI");
+    glXSwapIntervalEXT  = (int (*)(Display*, GLXDrawable drawable, int))glGetProcAddress("glXSwapIntervalEXT");
+    glXSwapIntervalMESA = (int (*)(int))glGetProcAddress("glXSwapIntervalMESA");
+    glXSwapIntervalSGI  = (int (*)(int))glGetProcAddress("glXSwapIntervalSGI");
 
     if(auto glXCreateContextAttribs = (auto (*)(Display*, GLXFBConfig, GLXContext, int, const int*) -> GLXContext)glGetProcAddress("glXCreateContextAttribsARB")) {
       int attributes[] = {
@@ -276,7 +276,7 @@
       return false;
     }
 
-    if(glXSwapInterval) glXSwapInterval(self.blocking);
+    glXSwapInterval(self.blocking);
 
     //read attributes of frame buffer for later use, as requested attributes from above are not always granted
     int value = 0;
@@ -312,7 +312,17 @@
 
   bool _ready = false;
 
-  auto (*glXSwapInterval)(int) -> int = nullptr;
+  auto glXSwapInterval(int blocking) -> int {
+    //note that the ordering is very important! MESA declares SGI, but the SGI function does nothing
+    if(glXSwapIntervalEXT  && glXSwapIntervalEXT(_display, glXGetCurrentDrawable(), blocking)) return 1;
+    if(glXSwapIntervalMESA && glXSwapIntervalMESA(blocking)) return 1;
+    if(glXSwapIntervalSGI  && glXSwapIntervalSGI(blocking)) return 1;
+    return 0;
+  }
+
+  auto (*glXSwapIntervalMESA)(int) -> int = nullptr;
+  auto (*glXSwapIntervalSGI)(int) -> int = nullptr;
+  auto (*glXSwapIntervalEXT)(Display *dpy, GLXDrawable drawable, int interval) -> int = nullptr;
 
   Display* _display = nullptr;
   u32 _monitorX = 0;
```

**Comment by Yasand123** [comment]
> Can you check if the following patch fixes the issue for you?

That fixed the issue. Thanks!

**Comment by LukeUsher** [comment]
Fixed in #[5636e2b](https://github.com/ares-emulator/ares/commit/5636e2b0f6a47dcf3b49ef9968d2bc874a0e2223)
